arangod/RocksDBEngine/RocksDBHotBackup.cpp

```diff
@@ -520,6 +523,34 @@ void RocksDBHotBackupCreate::executeCreate() {
       retVal = TRI_RenameFile(dirPathTemp.c_str(), dirPathFinal.c_str(), &systemError, &errors);
       _success = (TRI_ERROR_NO_ERROR == retVal);
     } // if
+
+    // write encrypted agency dump if available
+    if (_success && !_agencyDump.isNone()) {
+      std::string agencyDumpFileName = dirPathFinal + TRI_DIR_SEPARATOR_CHAR + "agency.json";
+
+#ifdef USE_ENTERPRISE
+      auto encryptFeature = application_features::ApplicationServer::lookupFeature<EncryptionFeature>("Encryption");
+
+      if (encryptFeature != nullptr) {
+        int fd = TRI_TRACKED_CREATE_FILE(agencyDumpFileName.c_str(), O_WRONLY | O_CREAT | O_TRUNC | TRI_O_CLOEXEC,
+                                     S_IRUSR | S_IWUSR | S_IRGRP);
+        if (fd != -1) {
+          TRI_DEFER(TRI_TRACKED_CLOSE_FILE(fd));
+
+          std::string json = _agencyDump.toJson();
```

Comment:
This code is using VPack.  You need to assume it will throw.  Yes, I just looked and toJson() leads to a code path within "Dumper" object that throws.  Not likely, but this is just one of those things.  Please put a try/catch around all code using _agencyDump.

Reply:
It is actually not so unlikely, if internals cannot be resolved.